### PR TITLE
fix: Calendar View 

### LINKED ISSUE #10428: [CAL-2266] Slots are open on wrong day in weekly view
### Issue Summary

Weekly view shows slots open on wrong day. 

Screenshot 2023-07-27 at 16 12 51

<sub>[CAL-2266](https://linear.app/calcom/issue/CAL-2266/slots-are-open-on-wrong-day-in-weekly-view)</sub>

cc @sean-brydon

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

Fixes https://github.com/calcom/cal.com/issues/10428

TODO

- [x] Fix type errors

## How to test this?

1) Create availability in IST Timezone ( 9 -5 , mon-fri except wed)
2) Set your system's time to New York
3) go to booking page and select weekly view to verify if dates and slots are correct

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: packages/features/bookings/Booker/components/LargeCalendar.tsx [hunk 1]
```diff
@@ -22,21 +27,25 @@ export const LargeCalendar = ({ extraDays }: { extraDays: number }) => {
...
     if (!schedule.data.slots) return availableTimeslots;
 
     for (const day in schedule.data.slots) {
       availableTimeslots[day] = schedule.data.slots[day].map((slot) => ({
-        // First formatting to LLL and then passing it to date prevents toDate()
-        // from changing the timezone to users local machine (instead of itmezone selected in UI dropdown)
-        start: new Date(dayjs(slot.time).utc().tz(timezone).format("LLL")),
-        end: new Date(dayjs(slot.time).utc().tz(timezone).add(eventDuration, "minutes").format("LLL")),
+        start: dayjs(slot.time).toISOString(),
+        end: dayjs(slot.time).add(eventDuration, "minutes").toISOString(),
```
Comment (Udit-takkar) [inline packages/features/bookings/Booker/components/LargeCalendar.tsx:37]: monthly view also stores available slots in ISOString

**Review by CarinaWolli — DISMISSED** [review]
The start day of the of the week is wrong, it always shows the day before the picked day: 
Screenshot 2023-08-03 at 09 40 11

File: packages/features/calendars/DatePicker.tsx [hunk 2]
```diff
@@ -169,7 +169,10 @@ const Days = ({
   return (
     <>
       {days.map((day, idx) => (
-        <div key={day === null ? `e-${idx}` : `day-${day.format()}`} className="relative w-full pt-[100%]">
+        <div
+          data-test={day?.format()}
```
[the remark this answers is not in the document]
Comment (Udit-takkar) [reply packages/features/calendars/DatePicker.tsx:173]: it's not required anywhere. i was checking the format

**Comment by Udit-takkar** [comment]
@CarinaWolli so the only issue pending is that you see one previous day but are slots are in correct days column?

**Comment by CarinaWolli** [comment]
> @CarinaWolli so the only issue pending is that you see one previous day but are slots are in correct days column?

Yes slots are showing on the correct days now 👏🏻

**Comment by sean-brydon** [comment]
CleanShot 2023-08-07 at 10 04 21@2x
Issue is fixed - noticed another thing. When hovering / selecting a slot we get MY timezone and not the timezone of i have selected in the booker

**Comment by Udit-takkar** [comment]
Screenshot 2023-08-08 at 3 26 54 PM

Everything should be fixed now. Tested on multiple timezone

**Review by sean-brydon — APPROVED** [review]
CleanShot 2023-08-09 at 09 37 51@2x
Seems to be working for me :) Great job! I know these TZ issues are a pain in the ass to work with!

**Pull request merged**